### PR TITLE
Ft testing mocha 187303859

### PR DESCRIPTION
**Pull request by RWEMAREMY**
#### What does this PR do?

testing with mocha locally

#### Description of Task to be completed?

- Using mocha with chai in testing

#### How should this be manually tested?

- use **npm run test** to test

#### Any background context you want to provide?

-dummy data are provided before having the real data ,use real data while testing and lets keep it on track of 85% and above
#### What are the relevant pivotal tracker/Trello stories?

- id of the project
#187303859

#### Questions:

### CODE REVIEW

**Review by musabehonore — APPROVED** [review]
I think this fantastic

**Review by Mukunzijames — APPROVED** [review]

**Review by RWEMAREMY — COMMENTED** [review]
done

File: src/config/config.js [hunk 1]
```diff
@@ -10,27 +10,35 @@ const getDatabaseConfig = () => {
   switch (env) {
     case 'test':
       return {
-        database: process.env.TEST_DATABASE_NAME ?? '',
-        username: process.env.TEST_DATABASE_USER ?? '',
-        password: process.env.TEST_DATABASE_PASSWORD ?? '',
-        host: process.env.TEST_DATABASE_HOST ?? '',
-        port: process.env.TEST_DATABASE_PORT ?? '',
+
+        database: process.env.TEST_DATABASE_NAME ?? "",
```
[the remark this answers is not in the document]
Comment (TuyishimireEric) [reply src/config/config.js:14]: Please resolve these errors, you may replace this line `database: process.env.TEST_DATABASE_NAME ?? "",` to `database: process.env.TEST_DATABASE_NAME ?? '',` 🙏

File: src/config/config.js [hunk 2]
```diff
@@ -10,27 +10,35 @@ const getDatabaseConfig = () => {
...
+        username: process.env.TEST_DATABASE_USER ?? "",
+        password: process.env.TEST_DATABASE_PASSWORD ?? "",
```
[the remark this answers is not in the document]
Comment (TuyishimireEric) [reply src/config/config.js:16]: Please `Rebase` to update your branch and resolve these **linter errors**, 🙏 

**Steps Perform the Rebase**
1. Checkout to the `develop` branch
2. Pull the updates from GitHub by running `git pull`
3. Checkout to your feature branch
4. start git rebase with `git rebase develop`
5. resolve the conflicts
6. continue to rebase with `git add .` and then `git rebase --continue`
7. save the changes 

These steps will update your **feature branch**, you have to run `npm i` to update `node_modules`,
please run `npm run lint:fix` and make sure that all **linter errors are fixed**

test your codes, 👍

[the remark this answers is not in the document]
Comment (TuyishimireEric) [reply src/config/config.js:16]: > Please `Rebase` to update your branch and resolve these **linter errors**, 🙏
> 
> **Steps Perform the Rebase**
> 
> 1. Checkout to the `develop` branch
> 2. Pull the updates from GitHub by running `git pull`
> 3. Checkout to your feature branch
> 4. start git rebase with `git rebase develop`
> 5. resolve the conflicts
> 6. continue to rebase with `git add .` and then `git rebase --continue`
> 7. save the changes
> 
> These steps will update your **feature branch**, you have to run `npm i` to update `node_modules`, please run `npm run lint:fix` and make sure that all **linter errors are fixed**
> 
> test your codes, 👍

After performing these steps the errors will be resolved

File: .nyc_output/processinfo/1d54fa34-f1f2-4c75-ac34-f17f9f74021a.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"parent":null,"pid":14836,"argv":["C:\\Program Files\\nodejs\\node.exe","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\node_modules\\mocha\\bin\\mocha.js","--require","ts-node/register","test/server-run.ts","--coverage"],"execArgv":[],"cwd":"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be","time":1711631797581,"ppid":9264,"coverageFilename":"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\.nyc_output\\1d54fa34-f1f2-4c75-ac34-f17f9f74021a.json","externalId":"","uuid":"1d54fa34-f1f2-4c75-ac34-f17f9f74021a","files":["C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\app.ts","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\routers\\index.ts","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\routers\\dummyRoute.ts","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\db.ts","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\config\\config.js","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\models\\index.ts","C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\models\\dummy.ts"]}
```
Comment (fab-ryan) [inline .nyc_output/processinfo/1d54fa34-f1f2-4c75-ac34-f17f9f74021a.json:1]: Please put this in gitignore

File: .nyc_output/processinfo/index.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"processes":{"1d54fa34-f1f2-4c75-ac34-f17f9f74021a":{"parent":null,"children":[]}},"files":{"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\app.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\routers\\index.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\routers\\dummyRoute.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\db.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\config\\config.js":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\models\\index.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"],"C:\\Users\\REMY\\Desktop\\ANDELA\\alpha-ec-be\\src\\database\\models\\dummy.ts":["1d54fa34-f1f2-4c75-ac34-f17f9f74021a"]},"externalIds":{}}
```
Comment (fab-ryan) [inline .nyc_output/processinfo/index.json:1]: same as here

File: coverage/lcov-report/base.css [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    color:#333;
+}
+.small { font-size: 12px; }
+*, *:after, *:before {
+  -webkit-box-sizing:border-box;
+     -moz-box-sizing:border-box;
+          box-sizing:border-box;
+  }
+h1 { font-size: 20px; margin: 0;}
+h2 { font-size: 14px; }
```
Comment (fab-ryan) [inline coverage/lcov-report/base.css:17]: a very output of test must be in gitignore

File: src/app.ts [hunk 6]
```diff
@@ -11,6 +11,10 @@ const app = express();
 app.use(cors());
 app.use(router);
 
+
+export default app;
+
```
Comment (fab-ryan) [inline src/app.ts:16]: why this export app???

File: src/config/config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fab-ryan) [reply src/config/config.js:14]: Please work on this comment

File: src/config/config.js [hunk 7]
```diff
@@ -10,27 +10,35 @@ const getDatabaseConfig = () => {
...
-        host: process.env.DATABASE_HOST ?? '',
-        port: process.env.DATABASE_PORT ?? '',
+        database: process.env.DATABASE_NAME ?? "",
+        username: process.env.DATABASE_USER ?? "",
+        password: process.env.DATABASE_PASSWORD ?? "",
+        host: process.env.DATABASE_HOST ?? "",
+        port: process.env.DATABASE_PORT ?? "",
+        dialect: "postgres",
+
+
```
Comment (fab-ryan) [inline src/config/config.js:41]: why the addition of those line you didn't rebase

File: src/controllers/demo.ts
Comment (fab-ryan) [inline src/controllers/demo.ts:1]: why creating this file we have our controller

File: src/controllers/dummyController.ts [hunk 8]
```diff
@@ -10,3 +10,8 @@ export const dummyRequest = async (req: Request, res: Response) => {
     res.status(500).json(err);
   }
 };
+export function addition (a: number, b: number): number {
+  return a + b;
+}
```
Comment (fab-ryan) [inline src/controllers/dummyController.ts:15]: Please don't change someone else codes

File: src/routers/dummyRoute.ts [hunk 9]
```diff
@@ -1,5 +1,8 @@
-import { Router } from 'express';
-import { dummyRequest } from '../controllers/dummyController';
+
+import  dummyRequest  from "../controllers/dummyController";
+import{createServer} from "../controllers/demo"
+import { Router } from "express";
+
```
Comment (fab-ryan) [inline src/routers/dummyRoute.ts:6]: same as here

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Please work on comment @TuyishimireEric  and I asked you

**Pull request closed without merging**